src/generated/xkb.rs

```diff
@@ -7085,22 +6722,30 @@ impl SelectEventsAux {
         self
     }
 }
-pub fn select_events<'c, Conn>(conn: &'c Conn, device_spec: DeviceSpec, affect_which: u16, clear: u16, select_all: u16, affect_map: u16, map: u16, details: &SelectEventsAux) -> Result<VoidCookie<'c, Conn>, ConnectionError>
+
+pub fn select_events<'c, Conn, A, B, C, D, E>(conn: &'c Conn, device_spec: DeviceSpec, affect_which: A, clear: B, select_all: C, affect_map: D, map: E, details: &SelectEventsAux) -> Result<VoidCookie<'c, Conn>, ConnectionError>
 where
     Conn: RequestConnection + ?Sized,
+    A: Into<u16>,
+    B: Into<u16>,
+    C: Into<u16>,
+    D: Into<u16>,
+    E: Into<u16>,
 {
     let extension_information = conn.extension_information(X11_EXTENSION_NAME)?
         .ok_or(ConnectionError::UnsupportedExtension)?;
-    let details_bytes = details.serialize();
+    let affect_which: u16 = affect_which.into();
+    let clear: u16 = clear.into();
+    let select_all: u16 = select_all.into();
+    let affect_map: u16 = affect_map.into();
+    let map: u16 = map.into();
     let length_so_far = 0;
     let device_spec_bytes = device_spec.serialize();
     let affect_which_bytes = affect_which.serialize();
     let clear_bytes = clear.serialize();
     let select_all_bytes = select_all.serialize();
     let affect_map_bytes = affect_map.serialize();
     let map_bytes = map.serialize();
-    assert_eq!((affect_which) & ((!(clear)) & (!(select_all))), details.value_mask(),
-        "Expression (affect_which) & ((!(clear)) & (!(select_all))) must match present values of 'details' argument (details.value_mask())");
```

Comment:
I think I haven't stumbled upon this one before. Another expression that needs to be `assert_eq!`'ed.

Reply:
Fixed